File: src/main/java/ru/arc/hooks/AEHook.java
package ru.arc.hooks;

import net.advancedplugins.ae.api.AEAPI;
import net.advancedplugins.ae.api.EnchantApplyEvent;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.Listener;
import org.bukkit.event.enchantment.EnchantItemEvent;
import org.bukkit.inventory.ItemStack;

import java.util.HashMap;

public class AEHook implements Listener {

    @EventHandler
    public void aeApply(EnchantApplyEvent event) {
        if (HookRegistry.sfHook != null && HookRegistry.sfHook.isSlimefunItem(event.getItem())) event.setCancelled(true);
    }
    
    @EventHandler(ignoreCancelled = true, priority = EventPriority.HIGHEST)
    public void onEnchantTable(EnchantItemEvent event) {
        ItemStack item = event.getItem();
        boolean slimefunItem = HookRegistry.sfHook.isSlimefunItem(item);
        ItemStack result = event.getItem();
        if (slimefunItem) {
            HashMap<String, Integer> enchantmentsOnItem = AEAPI.getEnchantmentsOnItem(item);
            for (String enchantment : enchantmentsOnItem.keySet()) {
                result = AEAPI.removeEnchantment(result, enchantment);
            }
        }
        if (result != item) {
            event.setItem(result);
        }
    }

}

File: src/main/kotlin/ru/arc/hooks/AEHook.kt
package ru.arc.hooks

import net.advancedplugins.ae.api.AEAPI
import net.advancedplugins.ae.api.EnchantApplyEvent
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.Listener
import org.bukkit.event.enchantment.EnchantItemEvent

class AEHook : Listener {

    @EventHandler
    fun aeApply(event: EnchantApplyEvent) {
        if (HookRegistry.sfHook?.isSlimefunItem(event.item) == true) event.isCancelled = true
    }

    @EventHandler(ignoreCancelled = true, priority = EventPriority.HIGHEST)
    fun onEnchantTable(event: EnchantItemEvent) {
        val item = event.item
        val sfHook = HookRegistry.sfHook ?: return
        if (!sfHook.isSlimefunItem(item)) return
        var result = item
        val enchantments = AEAPI.getEnchantmentsOnItem(item)
        for (enchantment in enchantments.keys) {
            result = AEAPI.removeEnchantment(result, enchantment)
        }
        if (result !== item) event.item = result
    }
}
